### PR TITLE
Move apollo-cache-inmemory to a regular dependency

### PR DESCRIPTION
**Pull request by astorije**
https://github.com/apollographql/react-apollo/blob/master/src/test-utils.tsx#L4 depends on `apollo-cache-inmemory`. When using the following in our tests:

```tsx
import { MockedProvider, MockedResponse } from 'react-apollo/test-utils';
```

The test suite fails with `Cannot find module 'apollo-cache-inmemory' from 'test-utils.js'`.

### Checklist:

* [ ] If this PR is a new feature, please reference an issue where a consensus about the design was reached (not necessary for small changes)
* [ ] Make sure all of the significant new logic is covered by tests
* [ ] If this was a change that affects the external API used in GitHunt-React, update GitHunt-React and post a link to the PR in the discussion.

### CODE REVIEW

**Comment by fbartho** [comment]
I don't believe this is a good idea. `test-utils` appears to be a utility for writing unit tests, react-apollo doesn't otherwise depend on apollo-cache-inmemory, does it?

**Comment by astorije** [comment]
The problem is that, as soon as you import/require `test-utils` in your tests, assuming your `package.json` depends on `react-apollo`, then as shown above `apollo-cache-inmemory` cannot be found.

**Comment by fbartho** [comment]
@astorije if you import test-utils, then you should also add `apollo-cache-inmemory` as a dependency to your project, no?

**Comment by astorije** [comment]
@fbartho, I don't think we should, because it's not something the consuming app's tests depend on, it's what a dependency it uses depends on. In fact, it's not a `peerDependency` nor is it mentioned at https://www.apollographql.com/docs/guides/testing-react-components.html so I actually think it should be a regular dependency.

**Comment by fbartho** [comment]
Let me rephrase my perspective: as I understand, none of the core code of this package depends on ‘inmemory; I see test utiles as a “sub package”. And react-apollo is commonly used with ‘inmemory, therefore this sub-package has tools that depend on it. Refactor proposals that make more sense to me over adding inmemory as a dep: change test-utils into a separate package, or move just the inmemory tests out, or move them into docs / code snippets.

**Comment by astorije** [comment]
@fbartho, I understand what you mean, but while I agree there are solutions for longer-term, at the moment tests on a consuming project will simply break if said project does not directly depend on `apollo-cache-inmemory`, which this package does not specify.

So the immediate fix is to either make it a `peerDependency` or add it to the `dependencies` of this package. I'd argue the latter is better because it doesn't force consumers to depend on an extra package on their end if the structure ends up changing (at which point, consumers will have forgotten about it and `apollo-cache-inmemory` will live in their `package.json`, unused, forever).

Does this make sense? 🙂

**Comment by rosskevin** [comment]
@fbartho is correct, test-utils are provided as part of the package to assist with testing and require the downstream project to setup devDependencies to support it.  These are not runtime resources and should never be treated as such.  The downstream project should be free to use whatever they choose in runtime, therefore adding to `dependencies` is a non-starter.

Similarly, this is not a `peerDependency`.  `peerDependency` has runtime implications and test-utils are explicitly intended for testing and to be pruned from runtime bundles.

**Comment by astorije** [comment]
Alright then, thanks for your input @rosskevin!
Maybe worth adding a note at https://www.apollographql.com/docs/react/recipes/testing.html that importing `react-apollo/test-utils` requires having `apollo-cache-inmemory` as a `devDependency` then?
